tests/test_butler.py

```diff
@@ -788,6 +788,31 @@ def testFileLocations(self):
         self.assertTrue(os.path.exists(os.path.join(self.dir1, "gen3.sqlite3")))
 
 
+class ButlerMakeRepoOutfileTestCase(PosixDatastoreButlerTestCase):
+    """Test that a config file created by makeRepo outside of repo works."""
+
+    def setUp(self):
+        self.root = tempfile.mkdtemp(dir=TESTDIR)
+        self.root2 = tempfile.mkdtemp(dir=TESTDIR)
+
+        self.tmpConfigFile = os.path.join(self.root, "different.yaml")
+        Butler.makeRepo(self.root, config=Config(self.configFile),
+                        outfile=self.tmpConfigFile)
+        print(f"YAML written to {self.tmpConfigFile}")
+        c = Config(self.tmpConfigFile)
+        print(c)
```

Comment:
Printing the config isn't really a test. Can you replace this with an assertion (e.g., that the recovered config is equal aside from the `root` field)?

Reply:
It seems that there are two more cases that could be tested: making `outfile` a directory (if this is not allowed, please say so in the documentation) and making it  a URI.

Reply:
Those print statements are some debugs that should not have been committed.

Reply:
Those two options (URI and directory only) do work but now I'm wondering about testing them since one put/get should be a sufficient test whereas at the moment I'm rerunning every butler test (because it's simpler that way). I may have to stop inheriting from the posix datastore test and do a one of test just for makeRepo outfile.